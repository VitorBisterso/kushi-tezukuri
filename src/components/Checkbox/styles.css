.checkbox-container > input {
  display: none;
}

.checkbox-container {
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  width: 18px;
  height: 18px;
  background-color: var(--COLOR-WHITE);
  border: 1px solid var(--COLOR-GOLD);
  border-radius: var(--BORDER_RADIUS-SMALLEST);
  transition: all 0.2s linear;
}

.checkbox-filled {
  display: none;
  width: 12px;
  height: 12px;
  border-radius: 2px;
  background-color: var(--COLOR-GOLD);
}

.checkbox-container > input:checked ~ .checkbox-filled {
  display: block;
}
