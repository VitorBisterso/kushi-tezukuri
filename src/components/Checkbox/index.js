import React from 'react'
import PropTypes from 'prop-types'

import './styles.css'

const Checkbox = ({ value, onChange }) => (
  // eslint-disable-next-line jsx-a11y/label-has-associated-control
  <label className="checkbox-container">
    <input type="checkbox" value={value} onClick={onChange} />
    <div className="checkbox-filled" />
  </label>
)

Checkbox.propTypes = {
  value: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
}

export default Checkbox
