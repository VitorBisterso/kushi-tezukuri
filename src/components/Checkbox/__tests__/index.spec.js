import React from 'react'
import { shallow } from 'enzyme'

import Checkbox from '../index'

describe('Checkbox', () => {
  it('should render correctly', () => {
    const onChangeFn = jest.fn()
    const component = shallow(<Checkbox value={false} onChange={onChangeFn} />)

    expect(component).toMatchSnapshot()
  })
})
