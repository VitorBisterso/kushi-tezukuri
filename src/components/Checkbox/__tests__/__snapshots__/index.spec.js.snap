// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
<label
  className="checkbox-container"
>
  <input
    onClick={[MockFunction]}
    type="checkbox"
    value={false}
  />
  <div
    className="checkbox-filled"
  />
</label>
`;
